doc/architectural/compilation-and-development.md

```diff
@@ -3,43 +3,106 @@
 
 \author Martin Brain, Peter Schrammel
```

Comment:
@owen-jones-diffblue add your name to the author's list or remove it. In general I don't think file author lists make any sense.